### ISSUE TITLE
Nested List Items Are Not Filtering (Prisma Client)

### ISSUE THREAD

**Issue by marcaaron** [prisma/prisma#3120]
**Describe the bug**
Skip and First does not return accurate results from related sub selections. Instead of filtering the results the entire collection is returned. orderBy works fine. first, skip do not work. When filtering the type directly filtering works fine, just not with a nested query.

**from datamodel.prisma**
```
type User {
  id: ID! @unique
  username: String! @unique
  email: String! @unique
  password: String!
  logs: [Log!]! @relation(name:"UserLogs", onDelete: CASCADE)
  sets: [Set!]! @relation(name:"UserSets", onDelete: CASCADE)
}

type Log {
  id: ID! @unique
  created_at: DateTime!
  sets: [Set!]! @relation(name: "SetsOnLog", onDelete: CASCADE)
  user: User! @relation(name:"UserLogs", onDelete: SET_NULL)
}

type Set {
  id: ID! @unique
  created_at: DateTime!
  movement: Movement!
  reps: Int!
  weight: Float!
  unit: String!
  user: User! @relation(name:"UserSets", onDelete: SET_NULL)
  log: Log! @relation(name: "SetsOnLog", onDelete: SET_NULL)
}

```

**Expected behavior**
(first: 1, skip:0) should return the first log on user not all logs on user...

**Screenshots**

![screen shot 307](https://user-images.githubusercontent.com/32969087/45653640-994e4a00-ba8d-11e8-8e49-bf6f71df42a5.png)
![screen shot 308](https://user-images.githubusercontent.com/32969087/45653643-9bb0a400-ba8d-11e8-83a7-1068b9bf752d.png)

**Versions (please complete the following information):**
 - OS: OS X High Sierra
 - `prisma` CLI: prisma/1.17.0-beta.16 (darwin-x64) node-v9.3.0
 - "prisma-client-lib": "^0.2.58"

**Additional context**
 - deployed to heroku w/ postgres

I believe this to be a bug but perhaps I've missed something?

[repo here: ](https://github.com/marcaaron/lift-logr)

**Comment by marcaaron** [comment]
Ok I have a new theory... 
@relation(name: "SomeName") // works fine
@relation(name:"SomeName") // without a space will break filtering.
Does that sound right?

**Comment by marcaaron** [comment]
Just kidding. Must have been a fluke. Not working anymore...

**Comment by marcaaron** [comment]
Things are seeming very inconsistent in general... not quite entirely broken. But I'm very confused...

Anotha One.

![Uploading Screen Shot 310.png…]()

**Comment by do4gr** [comment]
Thanks for filing this bug and providing a reproduction. It seems to only affect Postgres and a fix should be coming out soon.

**Comment by ficolo** [comment]
I think this is the same as #3097

**Comment by marcaaron** [comment]
Oh good! Thanks! I thought I was losing my mind for a second :)

**Comment by do4gr** [comment]
We just released 1.16.4 which should fix this issue.
